inc/contenttemplates/parameters/parameterstypes/abstractparametertype.class.php

```diff
@@ -60,6 +62,40 @@ abstract class AbstractParameterType
    /**
     * To be defined in each subclasses, convert the parameter data into an array
     * that can be shared to the client side code as json and used for autocompletion.
+    *
+    * @return array
     */
    abstract public function compute(): array;
+
+   /**
+    * Field name for this parameter's documentation
+    *
+    * @return string
+    */
+   abstract function getDocumentationField(): string;
+
+   /**
+    * Label to use for this parameter's documentation
+    *
+    * @return string
+    */
+   abstract function getDocumentationLabel(): string;
+
+   /**
+    * Recommended usage (twig code) to use for this parameter's documentation
+    *
+    * @param string|null $parent
+    *
+    * @return string
+    */
+   abstract function getDocumentationUsage(?string $parent = null): string;
+
+   /**
+    * Reference to others parameters for this parameter's documentation
+    *
+    * @return AbstractParameters|null
+    */
+   abstract function getDocumentationReferences(): ?AbstractParameters;
```

Comment:
1) The `public` keyword is missing on all new methods.

2) This abstract class has no abstract logic, it is more an interface than an abstract class. Maybe method declarations should be moved into an interface.

3) I guess some abstract default logic can be produced, like 
```php
   public function __construct(string $key, string $label) {
      $this->key = $key;
      $this->label = $label;
   }

   public function getDocumentationField(): string {
      return $this->key;
   }

   public function getDocumentationLabel(): string {
      return $this->label;
   }
```

Reply:
Done in 136e4f2fdb, not exactly as suggested since I realized that `$this->label` should not be part of the abstraction as it is not used for object parameters.